module/VuFind/src/VuFind/Db/Service/ResourceService.php

```diff
@@ -138,4 +199,73 @@ public function getResourceById($id)
         );
         return $resource;
     }
+
+    /**
+     * Use a record driver to assign metadata to the current row.
+     *
+     * @param \VuFind\RecordDriver\AbstractBase $driver    The record driver
+     * @param \VuFind\Date\Converter            $converter Date converter
+     * @param \VuFind\Db\Entity\Resource        $resource  Resource entity
+     *
+     * @return \VuFind\Db\Entity\Resource
+     */
+    public function assignMetadata(
+        $driver,
+        \VuFind\Date\Converter $converter,
+        $resource
+    ) {
+        // Grab title -- we have to have something in this field!
+        $title = mb_substr(
+            $driver->tryMethod('getSortTitle'),
+            0,
+            255,
+            "UTF-8"
+        );
+        if (empty($title)) {
+            $title = $driver->getBreadcrumb();
+        }
+        $resource->setTitle($title);
+        // Try to find an author; if not available, just leave the default null:
+        $author = mb_substr(
+            $driver->tryMethod('getPrimaryAuthor'),
```

Comment:
We should make this `$driver->tryMethod('getPrimaryAuthor') ?? ''` to ensure that we always pass a string to `mb_substr`.